### PR TITLE
Add disclaimer about IP logging in the manual authentication plugin

### LINKED ISSUE #991: Add a reminder about IP address privacy to the manual mode client
People may decide to run, e.g. the manual authenticator from their laptop rather than from the server. This will work fine. However, the Let's Encrypt privacy policy states that we will retain and publish IP addresses associated with ACME validation requests, including requestor IP. This is unremarkable when running the client on the same machine where a cert will be deployed. When running the client on a personal machine, this behavior will be unexpected to anyone who hasn't read the privacy policy.

We should do a check that the resolved names being authorized point at the current machine. If they don't we should provide a notice to the user that their IP address will be published, along with a link to the privacy policy.

### PR DESCRIPTION
**Pull request by ArchimedesPi**
Fixes #991.
ip_logging_disclaimer
Here's [a terminal recording](https://asciinema.org/a/2xmxfq43nxb9a9muiu3q290z7)!

### CODE REVIEW

**Comment by kuba** [comment]
I really think we should have a flag to automate it (`--manual-ip-logging-ok` defined in `manual.py`) before merging the PR.

**Comment by ArchimedesPi** [comment]
There are some odd problems with testing I have to figure out first.

Liam Marshall
student, {software,hardware,electronics} engineer
@archimedespi on Keybase <https://keybase.io/archimedespi>

**Comment by bmw** [comment]
@kuba, I don't think adding a flag hurts, but keep in mind this is the `manual` plugin. It won't even work without user interaction unless you're in `test-mode` which @ArchimedesPi, properly handled.

**Comment by kuba** [comment]
I agree, LGTM.

File: letsencrypt/plugins/manual.py [hunk 1]
```diff
@@ -151,6 +160,10 @@ def _perform_single(self, achall):
             if self._httpd.poll() is not None:
                 raise errors.Error("Couldn't execute manual command")
         else:
+            if not zope.component.getUtility(interfaces.IDisplay).yesno(
+                    self.IP_DISCLAIMER, "Yes", "No"):
+                raise errors.Error("Must agree to IP logging to proceed")
```
Comment (bmw) [inline letsencrypt/plugins/manual.py]: I'd change this to `errors.PluginError` and write a test for it to fix [coverage](https://travis-ci.org/letsencrypt/letsencrypt/jobs/87273420#L838).

**Comment by bmw** [comment]
Other than my comment above, this LGTM. Thanks @ArchimedesPi! You said there are some odd problems with testing. Anything we should be aware of before merging?

**Comment by ArchimedesPi** [comment]
@bmw: Switched `Error` to `PluginError`, fixed some nasty mock problems. No longer WIP.

**Comment by ArchimedesPi** [comment]
This PR should also probably be squashed :smiley:

File: letsencrypt/plugins/manual_test.py [hunk 2]
```diff
@@ -104,6 +106,17 @@ def test_cleanup_test_mode_kills_still_running(self, mock_killpg):
         self.auth_test_mode.cleanup(self.achalls)
         mock_killpg.assert_called_once_with(1234, signal.SIGTERM)
 
+    @mock.patch("letsencrypt.plugins.manual.zope.component.getUtility")
+    @mock.patch("letsencrypt.plugins.manual.sys.stdout")
+    @mock.patch("acme.challenges.SimpleHTTPResponse.simple_verify")
+    @mock.patch("__builtin__.raw_input")
```
Comment (bmw) [inline letsencrypt/plugins/manual_test.py]: None of the patches are necessary besides the one on `getUtility`.

File: letsencrypt/plugins/manual_test.py [hunk 3]
```diff
@@ -104,6 +106,17 @@ def test_cleanup_test_mode_kills_still_running(self, mock_killpg):
...
+    def test_disagree_with_ip_logging(self, mock_raw_input, mock_verify, mock_stdout,
```
Comment (bmw) [inline letsencrypt/plugins/manual_test.py]: Nit: It'd be nice if this was moved up with the rest of the `perform` tests.

File: letsencrypt/plugins/manual_test.py [hunk 4]
```diff
@@ -104,6 +106,17 @@ def test_cleanup_test_mode_kills_still_running(self, mock_killpg):
...
+                                      mock_interaction):
+        # pylint: disable=unused-argument
```
Comment (bmw) [inline letsencrypt/plugins/manual_test.py]: After removing the unneeded mocks, this can be deleted.

**Comment by ArchimedesPi** [comment]
@bmw Ah, I thought there might be some magic going on with all the patches, so I left them. Off I go to remove them and move the test.

**Comment by bmw** [comment]
Kicked Travis due to Boulder failures: [log1](https://gist.github.com/31e46ef059c724f3b196), [log2](https://gist.github.com/020b48bc52053a0596c8).

LGTM. Thanks for the PR @ArchimedesPi.

**Pull request merged**